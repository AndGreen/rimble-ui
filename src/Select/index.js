import React from 'react'
import styled from 'styled-components'

import {
  borders,
  borderColor,
  borderRadius,
  boxShadow
} from 'styled-system'

import ExpandMore from 'rmdi/lib/ExpandMore'

const Wrapper = styled.span`
  & {
    position: relative;
  }

  > svg {
    position: absolute;
    right: 1rem;
    top: 0;
    bottom: 0;
    pointer-events: none;
    margin: auto;
  }
`

const StyledSelect = styled.select`
  appearance: none;
  font-size: 1rem;
  line-height: 48px;
  padding: 0 3rem 0 1rem;
  background: #FFF;

  ${borders}
  ${borderColor}
  ${borderRadius}
  ${boxShadow}

  &:hover {
    box-shadow: 0px 2px 6px rgba(0, 0, 0, 0.3);
  }
  &:focus {
    outline: none;
    border-color: ${props => props.theme.colors.primary};
  }
  &:active {
  }
  &:disabled {
    opacity: 0.4;
    pointer-events: none;
  }
`

const Select = (props) => (
  <Wrapper>
    <StyledSelect {...props}>
      {props.items.map(
        (item, i) => (<option key={i} value={item}>{item}</option>)
      )}
    </StyledSelect>
    <ExpandMore />
  </Wrapper>
)

Select.defaultProps = {
  borders: 1,
  borderColor: 'grey',
  borderRadius: 1,
  boxShadow: 1
}

Select.displayName = 'Select'

export default Select
