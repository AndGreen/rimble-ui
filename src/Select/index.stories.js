import React from 'react'

import { storiesOf } from '@storybook/react'

import Select from '../Select'

const items = [
  'Wandering Thunder',
  'Black Wildflower',
  'Ancient Paper',
]

storiesOf('Select', module)
  .add('default', () => (
    <Select items={items} />
  ))
