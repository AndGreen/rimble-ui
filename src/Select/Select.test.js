import Select from './'

describe('Select component sanity', () => {
  it('has name', () => {
    expect(Select.displayName).toBe('Select')
  })

  it('matches default snapshot', () => {
    const component = shallow(<Select items={['Wandering Thunder', 'Black Wildflower', 'Ancient Paper',]} />)
    expect(component).toMatchSnapshot()
  })
})
