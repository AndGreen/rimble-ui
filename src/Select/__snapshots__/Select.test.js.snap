// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component sanity matches default snapshot 1`] = `
<styled.span>
  <styled.select
    borderColor="grey"
    borderRadius={1}
    borders={1}
    boxShadow={1}
    items={
      Array [
        "Wandering Thunder",
        "Black Wildflower",
        "Ancient Paper",
      ]
    }
    theme={
      Object {
        "borders": Array [
          0,
          "1px solid",
        ],
        "colors": Object {
          "black": "#000e1a",
          "blue": "#007ce0",
          "copyColor": "#3F3D4B",
          "grey": "#CCC",
          "navy": "#004175",
          "primary": "#4E3FCE",
          "white": "#fff",
        },
        "fontSizes": Array [
          12,
          14,
          16,
          20,
          24,
          32,
          48,
          64,
        ],
        "fontWeights": Array [
          0,
          300,
          400,
          600,
          700,
        ],
        "fonts": Object {
          "sansSerif": "\\"Source Sans Pro\\", -apple-system, sans-serif",
          "serif": "athelas, georgia, times, serif",
        },
        "heights": Array [
          0,
          16,
          32,
          64,
          128,
          256,
        ],
        "lineHeights": Object {
          "copy": 1.5,
          "solid": 1,
          "title": 1.25,
        },
        "maxHeights": Array [
          0,
          16,
          32,
          64,
          128,
          256,
        ],
        "maxWidths": Array [
          16,
          32,
          64,
          128,
          256,
          512,
          768,
          1024,
          1536,
        ],
        "minHeights": Array [
          0,
          16,
          32,
          64,
          128,
          256,
        ],
        "minWidths": Array [
          0,
          16,
          32,
          64,
          128,
          256,
        ],
        "opacity": Object {
          "disabled": 0.4,
        },
        "radii": Array [
          "0",
          "4px",
          "8px",
          "16px",
        ],
        "shadows": Array [
          "0",
          "0px 2px 4px rgba(0, 0, 0, 0.1)",
          "0 7px 14px rgba(50,50,93,.1)",
        ],
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
        ],
        "width": Array [
          0,
          16,
          32,
          64,
          128,
          256,
        ],
      }
    }
  >
    <option
      key="0"
      value="Wandering Thunder"
    >
      Wandering Thunder
    </option>
    <option
      key="1"
      value="Black Wildflower"
    >
      Black Wildflower
    </option>
    <option
      key="2"
      value="Ancient Paper"
    >
      Ancient Paper
    </option>
  </styled.select>
  <ExpandMoreIcon
    size={24}
  />
</styled.span>
`;
